packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md

```diff
@@ -0,0 +1,67 @@
+
+Which JWT type that you use depends on the client authentication method configured in your OAuth 2.0 client application. See [Client Authentication Methods](/docs/reference/api/oidc/#client-authentication-methods) for more information on the supported methods and when to use them.
+
+> **Note:** To create a client application and specify either the `client_secret_jwt` or `private_key_jwt` authentication method, see the [Add OAuth 2.0 Client Application](/docs/reference/api/apps/#add-oauth-2-0-client-application) API reference section. To change the client authentication method of an existing app, see the [Update the Client Authentication Method](/docs/reference/api/apps/#update-the-client-authentication-method) API reference section.
+
+> **Tip:** Don't know which method is configured for your client app? You can quickly check the settings by doing a GET to `https://${yourOktaDomain}/api/v1/apps/${appId}`.
+
+## Gather claims information
+
+When you create a JWT assertion claim for client authentication (`client_secret_jwt` or `private_key_jwt`), gather claims information for the payload section of the JWT. Claims are statements about the entity, which is typically a user and any additional data. 
+
+| Claim    | Description                                                  | Type        |
+|----------|--------------------------------------------------------------|-------------|
+| aud      | Required. The full URL of the resource that you're trying to access using the JWT to authenticate. For example: `https://${yourOktaDomain}/oauth2/default/v1/token` | String  |
+| exp      | Required. The token expiration time in seconds since January 1, 1970 UTC (UNIX timestamp), for example, `1555594819`. This claim fails the request if the expiration time is more than one hour in the future or if the token is already expired.            | Integer     |
+| iss      | Required. The issuer of the token. This value must be the same as the `client_id` of the application that you are accessing.  | String      |
+| sub      | Required. The subject of the token. This value must be the same as the `client_id` of the application that you are accessing. | String       |
+| jti      | Optional. The unique token identifier. If you specify this parameter, the token can only be used once and, as a result, subsequent token requests won't succeed. | String    |
+| iat      | Optional. When the token was issued in seconds since January 1, 1970 UTC (UNIX timestamp), for example, `1555591219`. If specified, it must be a time before the request is received. | Integer     |
+
+## Build a JWT with a shared key
+
+If you configured your application to use the `client_secret_jwt` client authentication method, then you want to build a JWT that you sign with the `client_secret` using an HMAC SHA algorithm (HS256, HS384, or HS512).
+
+<StackSelector snippet="createclientsecretjwt"/>
```

Comment:
Do you want to put this stack selector before the nut-facts based on the new template?
Add `noSelector` here if you are moving the selector up top.

Reply:
Yup, I'll add a visible stack selector up top, and hide all the others. Thanks!